pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/StUnionAggregationFunction.java

```diff
@@ -0,0 +1,137 @@
+    super(expression);
+  }
+
+  @Override
+  public AggregationFunctionType getType() {
+    return AggregationFunctionType.ST_UNION;
+  }
+
+  @Override
+  public void accept(AggregationFunctionVisitorBase visitor) {
+    visitor.visit(this);
+  }
+
+  @Override
+  public AggregationResultHolder createAggregationResultHolder() {
+    return new ObjectAggregationResultHolder();
+  }
+
+  @Override
+  public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int maxCapacity) {
+    return new ObjectGroupByResultHolder(initialCapacity, maxCapacity);
+  }
+
+  @Override
+  public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    byte[][] bytesArray = blockValSetMap.get(_expression).getBytesValuesSV();
+    Geometry geometry = aggregationResultHolder.getResult();
+    for (int i = 0; i < length; i++) {
+      Geometry value = GeometrySerializer.deserialize(bytesArray[i]);
+      geometry = geometry == null ? value : geometry.union(value);
+    }
+    aggregationResultHolder.setValue(geometry);
+  }
+
+  @Override
+  public void aggregateGroupBySV(int length, int[] groupKeyArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    byte[][] bytesArray = blockValSetMap.get(_expression).getBytesValuesSV();
+    for (int i = 0; i < length; i++) {
+      int groupKey = groupKeyArray[i];
+      Geometry value = GeometrySerializer.deserialize(bytesArray[i]);
+      groupByResultHolder.setValueForKey(groupKey, groupByResultHolder.getResult(groupKey) == null ? value
+          : ((Geometry) groupByResultHolder.getResult(groupKey)).union(value));
+    }
+  }
+
+  @Override
+  public void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResultHolder groupByResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    byte[][] bytesArray = blockValSetMap.get(_expression).getBytesValuesSV();
+    for (int i = 0; i < length; i++) {
+      Geometry value = GeometrySerializer.deserialize(bytesArray[i]);
+      for (int groupKey : groupKeysArray[i]) {
+        groupByResultHolder.setValueForKey(groupKey, groupByResultHolder.getResult(groupKey) == null ? value
+            : ((Geometry) groupByResultHolder.getResult(groupKey)).union(value));
+      }
+    }
+  }
+
+  @Override
+  public Geometry extractAggregationResult(AggregationResultHolder aggregationResultHolder) {
+    return aggregationResultHolder.getResult();
+  }
+
+  @Override
+  public Geometry extractGroupByResult(GroupByResultHolder groupByResultHolder, int groupKey) {
+    return groupByResultHolder.getResult(groupKey);
+  }
+
+  @Override
+  public Geometry merge(Geometry intermediateResult1, Geometry intermediateResult2) {
+    return intermediateResult1.union(intermediateResult2);
+  }
+
+  @Override
+  public boolean isIntermediateResultComparable() {
+    return true;
```

Comment:
Geometry is comparable?

Reply:
yes, https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/Geometry.html#compareTo-java.lang.Object-

Reply:
This is not the same comparison logic as the final result, so we should put `false` here.
This is for ordering the union geometry for group-by queries, which will use the final result `ByteArray` order. It does not make a lot of sense to order them though

Reply:
I see. So the assumption is that the intermediate result comparable order must be the same as the final result.

Reply:
Yes, because they are for the same purpose